File: hw/tipantiza/u3/hw25Composite/CompositeJava/src/main/java/ec/edu/espe/compositejava/model/Supervisor.java

package ec.edu.espe.compositejava.model;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Alexander Tipantiza, The Softwarriors, @ESPE
 */

public abstract class Supervisor extends Employee {

    protected List<Employee> employees;

    public Supervisor(String name, String title) {
        super(name, title);
        employees = new ArrayList<>();
    }

    public void add(Employee employee) {
        employees.add(employee);
    }

    public void remove(Employee employee) {
        employees.remove(employee);
    }

    @Override
    public void display() {

        System.out.println(title + ": " + name);

        for (Employee employee : employees) {
            employee.display();
        }
    }
}

File: hw/tipantiza/u3/hw25Composite/Composite_py/model/supervisor.py
from model.employee import Employee

class Supervisor(Employee):

    def __init__(self, name, title):
        super().__init__(name, title)
        self.employees = []

    def add(self, employee):
        self.employees.append(employee)

    def remove(self, employee):
        self.employees.remove(employee)

    def display(self):
        print(f"{self.title}: {self.name}")
        for employee in self.employees:
            employee.display()